store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -381,6 +378,14 @@ public boolean load() {
         return result;
     }
 
+    public void loadCheckPoint() throws IOException {
```

Comment:
This implies this PR won't build CQ records into RocksDB CQ store for existing messages in commit log if they are previously indexed by file-based CQ store.

Reply:
Config will be error prune. Once config of existing classic nodes are changed inappropriately, things will be totally messed up

Reply:
是这样的，需要按照issuse里边的配置来进行原地升级，如果直接改storeType，旧的消息将消费不到了，cq升级有俩种方案，一种是broker启动的时候做转换，一种是双写一段时间然后切流，我们选择了后者，这里的原因在issue里边有写，为了稳定性和不阻塞broker启动考虑的

Yes, it is necessary to perform an in-place upgrade according to the configuration in the issue. If you directly change the storeType, old messages will not be consumable. There are two solutions for upgrading the consume queue (CQ): one is to perform the conversion when the broker starts, and the other is to double-write for a period of time and then switch the flow. We chose the latter, as explained in the issue, for reasons of stability and to avoid blocking the broker startup.

Reply:
Then it's necessary to add a check and abort logic to guard against this incident from happening.